drivers/clock_control/nrf_clock_calibration.c

```diff
@@ -224,7 +226,7 @@ void z_nrf_clock_calibration_init(struct device *dev)
 
 	if (CONFIG_CLOCK_CONTROL_NRF_CALIBRATION_MAX_SKIP != 0) {
 		temp_sensor =
-			device_get_binding(DT_INST_0_NORDIC_NRF_TEMP_LABEL);
+			device_get_binding(DT_INST_LABEL(0));
```

Comment:
I think that something like this should be used instead:
```
		temp_sensor = device_get_binding(
			DT_NODE_HAS_PROP(DT_INST(0, nordic_nrf_temp), label)
				? DT_PROP(DT_INST(0, nordic_nrf_temp), label)
				: "");
```

Reply:
Unless I'm mistaken, this will fail to compile, because DT_PROP(..., some_prop_that_doesn't_exist) is a build error.

Handling cases like this is part of the reason I proposed adding UTIL_OR() and UTIL_AND() (https://github.com/zephyrproject-rtos/zephyr/pull/23753). Maybe we can use those here.

Reply:
Right, my mistake. I should have proposed using `COND_CODE_1` instead.
But on second thought, it seems to make little sense to call `device_get_binding("")`. Perhaps it would be better then to conditionally compile the whole thing? I mean:
```
#if DT_NODE_HAS_PROP(DT_INST(0, nordic_nrf_temp), label)
	if (CONFIG_CLOCK_CONTROL_NRF_CALIBRATION_MAX_SKIP != 0) {
		temp_sensor = device_get_binding(
			DT_NODE_HAS_PROP(DT_INST(0, nordic_nrf_temp), label));
	}
#endif
```
`temp_sensor` would then just stay initialized to `NULL` when the "nordic,nrf-temp" node is not enabled. @nordic-krch What you think?

Reply:
I guess I like the second (make the whole binding fetch optional), but is there any Nordic clock device that is on a SOC that doesn't have a corresponding temperature capability?  If not, I think the original change is just fine: making it conditional suggests a configuration that doesn't exist.

Reply:
I think that's the case of nRF5340 - calibration of LFRC is supported but there is no TEMP peripheral available. And that's why `DT_INST_0_NORDIC_NRF_TEMP_LABEL` was defined to `""`. @nordic-krch Please, correct me if I'm wrong.

Reply:
TEMP is only on network core. It is actually recommended that calibration is done only by single core (network if running).

Reply:
I am interpreting this to mean that `temp_sensor` should be set to `NULL` if there is no node label `temp` in zephyr.dts.